server/plugin.ts

```diff
@@ -126,6 +127,15 @@ export class OpendistroSecurityPlugin
         core
       );
       core.http.registerAuth(auth.authHandler);
+    } else if (config.auth.type === 'proxy') {
```

Comment:
Non-block, mind reminder, add an enum for auth types.

Reply:
sure